gorm/converter.go

```diff
@@ -340,15 +345,27 @@ func (converter *DefaultFilteringConditionConverter) StringArrayConditionToGorm(
 	placeholder := ""
 	for _, str := range c.Values {
 		placeholder += "?, "
-		if val, err := converter.Processor.ProcessStringCondition(ctx, c.FieldPath, str); err == nil {
-			values = append(values, val)
+		if value, err = converter.Processor.ProcessStringCondition(ctx, c.FieldPath, str); err == nil {
+			values = append(values, value)
 			continue
 		}
 
 		values = append(values, str)
 	}
 
-	return fmt.Sprintf("(%s %s %s (%s))", dbName, neg, o, strings.TrimSuffix(placeholder, ", ")), values, assocToJoin, nil
+	switch c.Type {
+	case query.StringArrayCondition_IN:
+		o = "IN"
+
+		return fmt.Sprintf("(%s %s %s (%s))", dbName, neg, o, strings.TrimSuffix(placeholder, ", ")), values, assocToJoin, nil
+
+	case query.StringArrayCondition_ANYOF:
+		o = "&&"
+	case query.StringArrayCondition_ALLOF:
```

Comment:
The "IN" operator would naturally map to contains i think. https://www.postgresql.org/docs/9.1/functions-array.html#ARRAY-OPERATORS-TABLE . The major difference is that the source column in this case in an array instead of a string. So the more correct thing to to would be to detect the type of the source field instead of forcing the user to use a different operator because the source field is different.

Reply:
Do we have an example were we detect the source field as an array and add an operator? @Calebjh

Reply:
> instead of forcing the user to use a different operator because the source field is different.

```IN``` operator does not applies to source fields of array type. We need new operators for array type:
https://www.postgresql.org/docs/13/gin-builtin-opclasses.html